fe/fe-core/src/main/java/com/starrocks/connector/ObjectBasedUpdateArbitrator.java

```diff
@@ -33,8 +33,10 @@ public Map<String, Optional<HivePartitionDataInfo>> getPartitionDataInfos() {
         if (partitionLimit >= 0 && partitionLimit < partitionNames.size()) {
             partitionNameToFetch = partitionNames.subList(partitionNames.size() - partitionLimit, partitionNames.size());
         }
+        GetRemoteFilesRequest request =
+                GetRemoteFilesRequest.newBuilder().setPartitionNames(partitionNames).setCheckPartitionExistence(false).build();
```

Comment:
why not check partition exist

Reply:
I don't know. Because in old code, it does not check partition existence. So I keep it.

Reply:
In old code:
- it fetch partition according to names(not check partition existence).
- it fetch remote files under those partitions
- and check what's last modification time.

Reply:
But I think we have to check partition existence anyway. Because if some partition does not exists, then

> `remoteFileInfos.size() != parttiionNameToFetch.size()`

It will crash here.

![Uploading image.png…]()

Reply:
But I think we have to check partition existence anyway. Because if some partition does not exists, then

> `remoteFileInfos.size() != parttiionNameToFetch.size()`

It will crash here.

![image](https://github.com/user-attachments/assets/d3c62d21-cca2-4ade-9f01-0c1d5c4fc505)